### PR TITLE
feat: error message for missing map type data

### PR DESCRIPTION
**Pull request by ShmuelLotman**
Closes idpe #9574

Introduces the error message for missing lat/lon data when attempting to view map type

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

Screen Shot 2021-02-04 at 10 39 01 AM

### CODE REVIEW

**Comment by hercules-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 49 | :red_circle:
accessibility | 65 | :yellow_circle:
best-practices | 87 | :green_circle:
seo | 80 | :yellow_circle:
pwa | null | :question:

File: src/shared/utils/vis.ts [hunk 1]
```diff
@@ -376,3 +379,41 @@ export const getMainColumnName = (
...
+})()
+
+export const getLatLon = (table, index: number) => {
+  const cellId = getS2CellID(table, index)
+  if (cellId === null || cellId.length > HEX_DIGIT_PRECISION) {
+    return null
+  }
+  if (cellId === null || cellId.length > HEX_DIGIT_PRECISION) {
+    return null
+  }
```
Comment (hoorayimhelping) [inline src/shared/utils/vis.ts:410]: hey apparently this check is so important it has to happen TWICE in a ROW!

File: src/shared/utils/vis.ts [hunk 2]
```diff
@@ -376,3 +379,41 @@ export const getMainColumnName = (
...
+  const column = table.getColumn('s2_cell_id')
+  if (!column) {
+    return null
+  }
+  const value = column[index]
+  if (typeof value !== 'string') {
+    return null
+  }
+  return value
+}
```
Comment (hoorayimhelping) [inline src/shared/utils/vis.ts:393]: this function is not very good, but it's fine for a demo. we'll fix it in post.

File: src/shared/utils/vis.ts [hunk 3]
```diff
@@ -376,3 +379,41 @@ export const getMainColumnName = (
...
+
+export const PRECISION_TRIMMING_TABLE = (() => {
+  const precisionTable = [BigInt(1)]
+  for (let i = 1; i < HEX_DIGIT_PRECISION + 1; i++) {
+    precisionTable[i] = precisionTable[i - 1] * BigInt(HEX_DIGIT_PRECISION)
+  }
+  return precisionTable
+})()
+
+export const getLatLon = (table, index: number) => {
```
Comment (hoorayimhelping) [inline src/shared/utils/vis.ts:403]: terrible function name. let's fix it quickly.

**Review by hoorayimhelping — APPROVED** [review]
pretty low quality code, but we'll fix it up in a fast follow:

https://github.com/influxdata/idpe/issues/9749

**Pull request merged**